proto/stride/stakeibc/trade_route.proto

```diff
@@ -39,15 +41,32 @@ message TradeRoute {
   // Currently Osmosis is the only trade chain so this is an osmosis pool id
   uint64 pool_id = 9;
 
-  // Spot price is a decimal ratio of the input to output denom as a string
-  string spot_price = 10;
+  // price of the output denom relative to the input denom
+  // this is update periodically via an ICQ
+  string host_token_price = 10 [
+    (cosmos_proto.scalar) = "cosmos.Dec",
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  // timestamp that the price was last updated
+  google.protobuf.Timestamp last_price_update_time = 11
+      [ (gogoproto.nullable) = false, (gogoproto.stdtime) = true ];
+
+  // max acceptable slippage from the trade, (e.g. 0.05 for 5%)
+  // the latest price and slippage determine the min output tokens from the
+  // trade
+  string max_slippage_percentage = 12 [
```

Comment:
Maybe splitting hairs, slippage is specifically loss from the action of the trade itself because of  shallow volume pool or something, but we also want to use this to defend against high variance price swings -- so the last seen price vs the current price if it is too large of a difference in the wrong direction -- that technically has nothing to do with slippage but we would still want to disallow the trade.  Also percentage makes me think the input would be "5" to mean 5% but you want "0.05" so I might rename this to "max_allowed_swap_loss" or something similar.

Reply:
yeah good points, I like "max_allowed_swap_loss"

Reply:
should we add a descriptor to indicate that it's a decimal? Maybe like "_rate" or "_ratio"?

Reply:
from Ethan on slack:
> yeah rate is good for this one I think